fortran/src/cudafor.f90

```diff
@@ -70,45 +70,29 @@ end function cudaSetDevice
 end interface cudaSetDevice
 !End cudaSetDevice
 
-!Start cudaMAlloc
-interface cudaMAlloc
-integer(c_int) function cudaMAlloc(devPtr, size) bind(c, name = "cudaMalloc")
+!Start cudaMalloc
+interface cudaMalloc
+integer(c_int) function cudaMalloc(devPtr, size) bind(c, name = "cudaMalloc")
 import :: c_int, c_size_t
 import :: c_devptr
 implicit none
 type(c_devptr) :: devPtr
 integer(c_size_t), value :: size
-end function cudaMAlloc
-end interface cudaMAlloc
-!End cudaMAlloc
+end function cudaMalloc
+end interface cudaMalloc
+!End cudaMalloc
 
 !Start cudaMemCpy
 interface cudaMemCpy
```

Comment:
Fortran sees `c_devptr` and `c_ptr` as different types and won't allow `c_devptr` to be passed to something that expects `c_ptr` and vice versa. The module and library will compile fine, but the tests don't compile for me with your patch with the compiler versions I mentioned.